cigaradvisor/blocks/article-teaser/article-teaser.js

```diff
@@ -0,0 +1,66 @@
+import { createOptimizedPicture } from '../../scripts/aem.js';
+
+function formatDate(originalDateString) {
+  const utcDateString = new Date((originalDateString - 25569) * 86400 * 1000);
+  const utcDate = new Date(utcDateString);
+  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+  const day = utcDate.getUTCDate();
+  const month = months[utcDate.getUTCMonth()];
+  const year = utcDate.getUTCFullYear().toString().slice(-2); // Get last two digits of the year
+  const formattedDate = `${day} ${month} ${year}`;
+  const dateTimeAttribute = `${utcDate.getUTCFullYear()}-${String(utcDate.getUTCMonth() + 1).padStart(2, '0')}-${String(utcDate.getUTCDate()).padStart(2, '0')}`;
+  return `${formattedDate}|${dateTimeAttribute}`;
+}
+
+async function fetchData(url) {
+  const resp = await fetch(url);
+  let jsonDataJson = '';
+  if (resp.ok) {
+    jsonDataJson = await resp.json();
+  }
+  return jsonDataJson.data;
+}
+
+export default async function decorate(block) {
+  const filterPath = block.querySelector('a').getAttribute('href');
+  block.textContent = '';
```

Comment:
There's no reason to clear out the block, as you replace its contents later.

Reply:
removed.